cli/config/configfile/file.go

```diff
@@ -285,19 +290,20 @@ func (configFile *ConfigFile) GetAllCredentials() (map[string]types.AuthConfig,
 		}
 	}
 
-	for registry := range configFile.CredentialHelpers {
-		helper := configFile.GetCredentialsStore(registry)
-		newAuths, err := helper.GetAll()
-		if err != nil {
-			return nil, err
-		}
-		addAll(newAuths)
-	}
 	defaultStore := configFile.GetCredentialsStore("")
 	newAuths, err := defaultStore.GetAll()
 	if err != nil {
 		return nil, err
 	}
 	addAll(newAuths)
+
+	// Auth configs from a registry-specific helper should override those from the default store.
+	for serverAddress := range configFile.CredentialHelpers {
```

Comment:
nit: `serverAdress` is a bit too vague as a var name. Can we keep `registry` or `registryAddr` (same in the tests)?

Reply:
LGTM besides this

Reply:
registryHostname? Done.